### ISSUE TITLE
selectedTo does not clear when new data is set

### ISSUE THREAD

**Issue by omrihar** [ratiw/vuetable-2#481]
I have a `vuetable` whose data depends on a selection from a list. When the list value changes, I change the data on the table (not through `apiMode`). I noticed that when the data changes and the rows with the value currently in `selectedTo` are no longer available, `selectedTo` is still populated. This could cause that the array contains values from different datasets.

I would recommend implementing some logic that only allows values that are in the current data model of the table. Currently I clear the array manually but it doesn't feel like a very elegant solution. What do you think?

**Comment by ratiw** [comment]
@omrihar That is true. I haven't used the Data mode that much, so I haven't thought about this. 

I think I could provide another function to help clearing out the populated `selectedTo`. But determining a proper logic to automatically clearing this will be a bit difficult as it might result in different behavior in the API mode. 

In API mode, Vuetable will keep all the selected values across pages and this is the default behavior. So, it is expected that you will have to manually clearing out the selected values in `selectedTo` during the `vuetable:loading` event if you do not want this behavior. In Data mode, it is also expected to be the same in order to keep it consistent. This, however, can be done easier if Vuetable has a function like `clearSelectedValues()` or sort of.

Will this be enough for your use case? Or do you have any suggestion that we can discuss? :)

**Comment by omrihar** [comment]
@ratiw, I think this would be enough. Actually, it does make sense to keep the values between loads of the data for the sake of pagination. Perhaps it would also be enough to have a method to retrieve only the part of the array that is also part of the current data model. This might also be relevant for API mode, in case the user just wants the values selected on the current page.
